tgctoolbox/settings.py

```diff
@@ -0,0 +1,107 @@
+import logging
+import os
+
+import yaml
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+BASE_FILES = [
+    os.path.join("configuration", "base.yaml"),
+    os.path.join("configuration", "local.yaml"),
+    os.path.join("configuration", "production.yaml"),
+]
+
+
+class Settings:
+    """Class to manage application settings."""
+
+    _instance = None
+
+    def __new__(cls, *args, **kwargs):
+        if cls._instance is None:
+            cls._instance = super().__new__(cls)
+        return cls._instance
+
+    def __init__(self, config_files=None, reload=False):
+        """
+        Initialize Settings object.
+
+        Args:
+            config_files (list): List of YAML configuration files.
+            reload (bool): Whether to reload settings.
+        """
+        if not hasattr(self, "_initialized") or reload:
+            self._settings = {}
+            config_files = config_files or BASE_FILES
+            self._load_from_yaml(config_files)
+            self._load_from_env()
+            self._initialized = True

```

Comment:
if all of the fields weren't filled up we should terminate the programme due to invalid settings and inform the user in appropiate way

Reply:
Resolved in https://github.com/TwoGuysCoding/TGCToolbox/commit/4c8ba0fffe80f07725e60d2540a28dce9ab00584